### PR TITLE
Add general edit modal fields

### PR DESCRIPTION
**Pull request by jessie-ross**
**This PR adds fields to the General Edit Modal.**

This one is pretty straight forward:

Screen Shot 2020-08-22 at 3 41 21 am

## How has this been tested?

- Automated tests that cover added/changed functionality
- Manual testing with mock data (locally)
- E2E testing for all of these to come at the end.

### CODE REVIEW

File: components/GeneralPanel.tsx [hunk 1]
```diff
@@ -66,7 +83,10 @@ function GeneralPanel({ experiment }: { experiment: ExperimentFull }) {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const [isEditing, setIsEditing] = useState<boolean>(false)
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (yanirs) [inline components/GeneralPanel.tsx:85]: Can some of these be removed now?

File: components/GeneralPanel.tsx [hunk 2]
```diff
@@ -91,12 +111,85 @@ function GeneralPanel({ experiment }: { experiment: ExperimentFull }) {
...
         <DialogTitle id='edit-experiment-general-form-dialog-title'>Edit Experiment: General</DialogTitle>
         <Formik initialValues={{ experiment: generalEditInitialExperiment }} onSubmit={onSubmitEdit}>
           {(formikProps) => (
             <form onSubmit={formikProps.handleSubmit}>
-              <DialogContent></DialogContent>
+              <DialogContent>
+                <div className={classes.row}>
+                  <Field
+                    component={TextField}
+                    id='experiment.p2Url'
```
Comment (yanirs) [inline components/GeneralPanel.tsx:123]: We don't allow edits to the P2 URL for non-staging experiments.

File: components/GeneralPanel.tsx [hunk 3]
```diff
@@ -91,12 +111,85 @@ function GeneralPanel({ experiment }: { experiment: ExperimentFull }) {
...
+                      shrink: true,
+                    }}
+                  />
+                </div>
+
+                <div className={classes.row}>
+                  <Field
+                    component={TextField}
+                    className={classes.datePicker}
+                    name='experiment.endDatetime'
```
Comment (yanirs) [inline components/GeneralPanel.tsx:159]: The experiment end datetime is only editable if the experiment is still running. Shall we disable/hide this field if it's not running?

Comment (jessie-ross) [reply components/GeneralPanel.tsx:159]: Hmm, knowing what's editable when is going to be a bit tricky for users.

Comment (jessie-ross) [reply components/GeneralPanel.tsx:159]: This is what I've done to it:

Screen Shot 2020-08-25 at 11 18 08 pm

Comment (yanirs) [reply components/GeneralPanel.tsx:141]: Looks good! 

This is a bit out of scope of this PR, but we should hide the `Edit` button on this panel if the experiment status is `staging`. If we don't, the "cannot be changed as the experiment has finished" text isn't necessarily true. This also applies to other edit buttons, so might be better to address separately.

**Review by yanirs — APPROVED** [review]

**Pull request merged**